x-pack/plugins/infra/server/routes/profiling/index.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { decodeOrThrow } from '@kbn/io-ts-utils';
+import { InfraProfilingRequestParamsRT } from '../../../common/http_api/profiling_api';
+import type { InfraBackendLibs } from '../../lib/infra_types';
+import { fetchProfilingFlamegraph } from './lib/fetch_profiling_flamechart';
+import { fetchProfilingStatus } from './lib/fetch_profiling_status';
+import { getProfilingDataAccess } from './lib/get_profiling_data_access';
+
+export function initProfilingRoutes({ framework, getStartServices, logger }: InfraBackendLibs) {
+  if (!Object.hasOwn(framework.plugins, 'profilingDataAccess')) {
+    logger.info(
+      "Skipping initialization of Profiling endpoints because 'profilingDataAccess' plugin is not available"
+    );
+    return;
+  }
+
+  framework.registerRoute(
+    {
+      method: 'get',
+      path: '/api/infra/profiling/status',
+      validate: false,
+    },
+    async (requestContext, request, response) => {
+      const coreRequestContext = await requestContext.core;
+      const infraRequestContext = await requestContext.infra;
+      const profilingDataAccess = await getProfilingDataAccess(getStartServices);
+
+      const profilingStatus = await fetchProfilingStatus(
+        profilingDataAccess,
+        coreRequestContext,
+        infraRequestContext
+      );
+
+      return response.ok({
+        body: profilingStatus,
+      });
+    }
+  );
+
+  framework.registerRoute(
+    {
+      method: 'post',
```

Comment:
Is it intentional that this path uses `Post` method? Shouldn't we use `Get`?

Reply:
Maybe it's an old habit at this point, but I usually avoid GET when there is request payload, at least in the past that was causing problems with some server implementations. But you're right that semantically GET fits better here, I'll probably switch to use URL params instead of body payload.

Reply:
Tried converting to GET with URL params, but the code becomes a bit cumbersome with trying to URL encode, it's a bit cleaner when using POST. If you don't mind, I'd leave it as it is considering that it's a common thing in Infra API to use POST pretty much everywhere. It seems like with the current implementation of `useHTTPRequest`, it's just easier to work with POST.